arm-wt-22k/src/hostmm_ng.c

```diff
@@ -0,0 +1,381 @@
+// hostmm_ng.c
+// This implements EAS Host Wrapper, using C runtime library, with better performance
+
+#include "eas_host.h"
+#include <fcntl.h>
+#include <limits.h>
+#include <pthread.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if defined (__linux__)
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+#elif defined (_WIN32)
+#include <io.h>
+#include <windows.h>
+#endif
+
+/* Only for debugging LED, vibrate, and backlight functions */
+#include "eas_report.h"
+
+#if __has_include(<byteswap.h>)
+#include <byteswap.h>
+#define bswap16 __bswap_16
+#define bswap32 __bswap_32
+#elif defined(_MSC_VER)
+#define bswap16 _byteswap_ushort
+#define bswap32 _byteswap_ulong
+#elif defined(__GNUC__) || defined(__clang__)
+#define bswap16 __builtin_bswap16
+#define bswap32 __builtin_bswap32
+#endif
+
+// https://stackoverflow.com/a/2103095
+// CC BY-SA 3.0
+enum {
+    O32_LITTLE_ENDIAN = 0x03020100ul,
+    O32_BIG_ENDIAN = 0x00010203ul,
+    O32_PDP_ENDIAN = 0x01000302ul, /* DEC PDP-11 (aka ENDIAN_LITTLE_WORD) */
+    O32_HONEYWELL_ENDIAN = 0x02030001ul /* Honeywell 316 (aka ENDIAN_BIG_WORD) */
+};
+
+static const union {
+    uint8_t bytes[4];
+    uint32_t value;
+} o32_host_order = { { 0, 1, 2, 3 } };
+
+#define O32_HOST_ORDER (o32_host_order.value)
+// --- end
```

Comment:
Use the CMake defined macro:

```suggestion
const EAS_BOOL O32_BIG_ENDIAN = 
#ifdef BIG_ENDIAN
    EAS_TRUE;
#else
    EAS_FALSE;
#endif
```